Source/WebCore/inspector/agents/InspectorCSSAgent.cpp

```diff
@@ -83,34 +83,43 @@ using namespace Inspector;
 
 enum ForcePseudoClassFlags {
     PseudoClassNone = 0,
-    PseudoClassHover = 1 << 0,
-    PseudoClassFocus = 1 << 1,
-    PseudoClassActive = 1 << 2,
-    PseudoClassVisited = 1 << 3
+    PseudoClassActive = 1 << 0,
+    PseudoClassHover = 1 << 1,
+    PseudoClassFocus = 1 << 2,
+    PseudoClassFocusVisible = 1 << 3,
+    PseudoClassFocusWithin = 1 << 4,
+    PseudoClassTarget = 1 << 5,
+    PseudoClassVisited = 1 << 6,
 };
 
 static unsigned computePseudoClassMask(const JSON::Array& pseudoClassArray)
 {
-    static NeverDestroyed<String> active(MAKE_STATIC_STRING_IMPL("active"));
-    static NeverDestroyed<String> hover(MAKE_STATIC_STRING_IMPL("hover"));
-    static NeverDestroyed<String> focus(MAKE_STATIC_STRING_IMPL("focus"));
-    static NeverDestroyed<String> visited(MAKE_STATIC_STRING_IMPL("visited"));
-    if (!pseudoClassArray.length())
-        return PseudoClassNone;
+    static NeverDestroyed<String> protocolActive(MAKE_STATIC_STRING_IMPL("active"));
+    static NeverDestroyed<String> protocolHover(MAKE_STATIC_STRING_IMPL("hover"));
+    static NeverDestroyed<String> protocolFocus(MAKE_STATIC_STRING_IMPL("focus"));
+    static NeverDestroyed<String> protocolFocusVisible(MAKE_STATIC_STRING_IMPL("focus-visible"));
+    static NeverDestroyed<String> protocolFocusWithin(MAKE_STATIC_STRING_IMPL("focus-within"));
+    static NeverDestroyed<String> protocolTarget(MAKE_STATIC_STRING_IMPL("target"));
+    static NeverDestroyed<String> protocolVisited(MAKE_STATIC_STRING_IMPL("visited"));
```

Comment:
Can we just make this an actual enum in the protocol that creates strings for us? It seems like we end up with 5 possible representations of the same value currently. The frontend's enum, the protocol's anonymous enum, these strings, the `ForcePseudoClassFlags` enum, and the `CSSSelector::PseudoClass*` enum (see my note below on possible getting rid of `ForcePseudoClassFlags` too).

Reply:
yeah ill move the anonymous enum to be named